src/Private/Pop-WindowsNotification.ps1

```diff
@@ -1,35 +1,52 @@
-Function Pop-Notification {
+Function Pop-WindowsNotification {
 
-    [CmdletBinding()]
+    [CmdletBinding(SupportsShouldProcess=$true)]
     param (
         [Parameter(Mandatory=$true, ValueFromPipeline=$true)]
         [string]
         $Body,
 
         [Parameter(Mandatory=$true)]
         [string]
-        $Title
+        $Title,
+
+        [Parameter()]
+        [string]
+        $Icon = "$PSScriptRoot\lib\alarm.png",
+
+        [Parameter()]
+        [switch]
+        $Silent
     )
 
+    if ($Silent) {
+        $SoundElement = '<audio silent="true" />'
+    } else {
+        $SoundElement = '<audio src="ms-winsoundevent:Notification.Default" />'
+    }
+
     $XmlString = @"
     <toast>
     <visual>
         <binding template="ToastGeneric">
         <text>$Title</text>
         <text>$Body</text>
-        <image src="$PSScriptRoot\lib\alarm.png" placement="appLogoOverride" hint-crop="circle" />
+        <image src="$Icon" placement="appLogoOverride" hint-crop="circle" />
         </binding>
     </visual>
-    <audio src="ms-winsoundevent:Notification.Default" />
+    $SoundElement
     </toast>
 "@
 
-    $AppId = '{1AC14E77-02E7-4E5D-B744-2EB1AE5198B7}\WindowsPowerShell\v1.0\powershell.exe'
+    $AppId = Get-WindowsAppId
+
     $null = [Windows.UI.Notifications.ToastNotificationManager, Windows.UI.Notifications, ContentType = WindowsRuntime]
     $null = [Windows.Data.Xml.Dom.XmlDocument, Windows.Data.Xml.Dom.XmlDocument, ContentType = WindowsRuntime]
     $ToastXml = [Windows.Data.Xml.Dom.XmlDocument]::new()
     $ToastXml.LoadXml($XmlString)
     $Toast = [Windows.UI.Notifications.ToastNotification]::new($ToastXml)
-    [Windows.UI.Notifications.ToastNotificationManager]::CreateToastNotifier($AppId).Show($Toast)
 
-}
\ No newline at end of file
+    if($PSCmdlet.ShouldProcess("running: CreateToastNotifier method with AppId $AppId and XML Payload: `r`n$XmlString")) {
```

Comment:
Normally I would call out the `r`n and say to use `Environment.NewLine` but we're promising that this code will get run only on Windows, so this is totally valid and makes sense. Just wanted to call out that I thought this :smile: